data-structures/src/data/structure/map/overlinkedlist/LinkedHashMap.java

```diff
@@ -0,0 +1,97 @@
+package data.structure.map.overlinkedlist;
+
+import data.structure.linkedlist.LinkedList;
+import data.structure.map.openaddressing.OpenAddressingMap;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
+public class LinkedHashMap<K, V> {
+    private static final int STARTING_CAPACITY = 12;
+
+    private int size;
+    private final Object[] items;
+
+    private class Entry {
+        final K key;
+        final V value;
+
+        Entry(final K key, final V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public LinkedHashMap() {
+        size = 0;
+        items = new Object[STARTING_CAPACITY];
+    }
+
+    @SuppressWarnings("unchecked")
+    public void put(final K key, final V value) {
+        requireNonNull(key, "The key must not be null!");
+
+        final int index = Math.abs(key.hashCode()) % STARTING_CAPACITY;
+
+        if (isNull(items[index])) {
+            items[index] = new LinkedList<Entry>();
+            ((LinkedList<Entry>)items[index]).add(new Entry(key, value));
+            size++;
+            return;
+        }
+
+        if (!isNull(items[index])) {

```

Comment:
There is no need for an `if` here, since you return from the previous one. Thus, if execution reaches this point, it is guaranteed that `items[index]` is non-null.